Bump psycopg2

python 3.12 support added in psycopg2 2.9.9

also maybe fixes for some disconnects?

```
2024-02-12 10:13:33,777 | matrix_reminder_bot.callbacks [ERROR] Unknown error while processing command:
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/matrix_reminder_bot/callbacks.py", line 113, in message
    await command.process()
  File "/usr/local/lib/python3.11/site-packages/matrix_reminder_bot/bot_commands.py", line 305, in process
    await self._remind_me()
  File "/usr/local/lib/python3.11/site-packages/matrix_reminder_bot/functions.py", line 99, in inner
    await command_func(self, *args, **kwargs)
  File "/usr/local/lib/python3.11/site-packages/matrix_reminder_bot/bot_commands.py", line 339, in _remind_me
    await self._remind(target=self.event.sender)
  File "/usr/local/lib/python3.11/site-packages/matrix_reminder_bot/bot_commands.py", line 297, in _remind
    self.store.store_reminder(reminder)
  File "/usr/local/lib/python3.11/site-packages/matrix_reminder_bot/storage.py", line 359, in store_reminder
    self._execute(
  File "/usr/local/lib/python3.11/site-packages/matrix_reminder_bot/storage.py", line 71, in _execute
    self.cursor.execute(args[0].replace("?", "%s"), *args[1:])
psycopg2.OperationalError: server closed the connection unexpectedly
	This probably means the server terminated abnormally
	before or while processing the request.

2024-02-12 10:14:04,152 | matrix_reminder_bot.callbacks [ERROR] Unknown error while processing command:
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/matrix_reminder_bot/callbacks.py", line 113, in message
    await command.process()
  File "/usr/local/lib/python3.11/site-packages/matrix_reminder_bot/bot_commands.py", line 330, in process
    await self._delete_reminder()
  File "/usr/local/lib/python3.11/site-packages/matrix_reminder_bot/functions.py", line 99, in inner
    await command_func(self, *args, **kwargs)
  File "/usr/local/lib/python3.11/site-packages/matrix_reminder_bot/bot_commands.py", line 530, in _delete_reminder
    reminder.cancel()
  File "/usr/local/lib/python3.11/site-packages/matrix_reminder_bot/reminder.py", line 181, in cancel
    self.store.delete_reminder(self.room_id, self.reminder_text)
  File "/usr/local/lib/python3.11/site-packages/matrix_reminder_bot/storage.py", line 388, in delete_reminder
    self._execute(
  File "/usr/local/lib/python3.11/site-packages/matrix_reminder_bot/storage.py", line 71, in _execute
    self.cursor.execute(args[0].replace("?", "%s"), *args[1:])
psycopg2.InterfaceError: cursor already closed
2024-02-12 10:15:00,555 | apscheduler.executors.default [ERROR] Job "Reminder._fire (trigger: date[2024-02-12 10:15:00 CET], next run at: 2024-02-12 10:15:00 CET)" raised an exception
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/apscheduler/executors/base_py3.py", line 30, in run_coroutine_job
    retval = await job.func(*job.args, **job.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/matrix_reminder_bot/reminder.py", line 145, in _fire
    self.cancel(cancel_alarm=False)
  File "/usr/local/lib/python3.11/site-packages/matrix_reminder_bot/reminder.py", line 181, in cancel
    self.store.delete_reminder(self.room_id, self.reminder_text)
  File "/usr/local/lib/python3.11/site-packages/matrix_reminder_bot/storage.py", line 388, in delete_reminder
    self._execute(
  File "/usr/local/lib/python3.11/site-packages/matrix_reminder_bot/storage.py", line 71, in _execute
    self.cursor.execute(args[0].replace("?", "%s"), *args[1:])
psycopg2.InterfaceError: cursor already closed
```

--- a/setup.py
+++ b/setup.py
@@ -38,7 +38,7 @@ setup(
         "readabledelta>=0.0.2",
         "apscheduler>=3.10.4",
         "pytz>=2024.1",
-        "arrow>=0.17.0",
+        "arrow>=1.3.0",
         "pretty_cron>=1.2.0",
     ],
     extras_require={

--- a/setup.py
+++ b/setup.py
@@ -33,7 +33,7 @@ setup(
     install_requires=[
         "matrix-nio[e2e]>=0.24.0",
         "Markdown>=3.5.2",
-        "PyYAML>=5.1.2",
+        "PyYAML>=6.0.1",
         "dateparser>=0.7.4",
         "readabledelta>=0.0.2",
         "apscheduler>=3.6.3",

--- a/setup.py
+++ b/setup.py
@@ -32,7 +32,7 @@ setup(
     packages=find_packages(exclude=["tests", "tests.*"]),
     install_requires=[
         "matrix-nio[e2e]>=0.24.0",
-        "Markdown>=3.1.1",
+        "Markdown>=3.5.2",
         "PyYAML>=5.1.2",
         "dateparser>=0.7.4",
         "readabledelta>=0.0.2",

--- a/setup.py
+++ b/setup.py
@@ -37,7 +37,7 @@ setup(
         "dateparser>=1.2.0",
         "readabledelta>=0.0.2",
         "apscheduler>=3.10.4",
-        "pytz>=2020.1",
+        "pytz>=2024.1",
         "arrow>=0.17.0",
         "pretty_cron>=1.2.0",
     ],

--- a/setup.py
+++ b/setup.py
@@ -42,7 +42,7 @@ setup(
         "pretty_cron>=1.2.0",
     ],
     extras_require={
-        "postgres": ["psycopg2>=2.8.5"],
+        "postgres": ["psycopg2>=2.9.9"],
         "dev": [
             "isort==5.13.2",
             "flake8==7.0.0",

--- a/setup.py
+++ b/setup.py
@@ -34,7 +34,7 @@ setup(
         "matrix-nio[e2e]>=0.24.0",
         "Markdown>=3.5.2",
         "PyYAML>=6.0.1",
-        "dateparser>=0.7.4",
+        "dateparser>=1.2.0",
         "readabledelta>=0.0.2",
         "apscheduler>=3.6.3",
         "pytz>=2020.1",

--- a/setup.py
+++ b/setup.py
@@ -36,7 +36,7 @@ setup(
         "PyYAML>=6.0.1",
         "dateparser>=1.2.0",
         "readabledelta>=0.0.2",
-        "apscheduler>=3.6.3",
+        "apscheduler>=3.10.4",
         "pytz>=2020.1",
         "arrow>=0.17.0",
         "pretty_cron>=1.2.0",

